### Importación de datos



In [2]:
import pandas as pd

url = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_1%20.csv"
url2 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_2.csv"
url3 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_3.csv"
url4 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_4.csv"

tienda = pd.read_csv(url)
tienda2 = pd.read_csv(url2)
tienda3 = pd.read_csv(url3)
tienda4 = pd.read_csv(url4)

tienda.head()

Producto Categoría del Producto    Precio  Costo de envío  \
0  Asistente virtual           Electrónicos  164300.0          6900.0   
1    Mesa de comedor                Muebles  192300.0          8400.0   
2      Juego de mesa               Juguetes  209600.0         15900.0   
3         Microondas      Electrodomésticos  757500.0         41000.0   
4   Silla de oficina                Muebles  335200.0         20200.0   

  Fecha de Compra         Vendedor Lugar de Compra  Calificación  \
0      16/01/2021      Pedro Gomez          Bogotá             4   
1      18/05/2022  Beatriz Morales        Medellín             1   
2      15/03/2021   Juan Fernandez       Cartagena             1   
3      03/05/2022   Juan Fernandez            Cali             4   
4      07/11/2020    Maria Alfonso        Medellín             5   

       Método de pago  Cantidad de cuotas       lat       lon  
0  Tarjeta de crédito                   8   4.60971 -74.08175  
1  Tarjeta de crédito                   4   6.25184 -75.56359  
2  Tarjeta de crédito                   1  10.39972 -75.51444  
3               Nequi                   1   3.43722 -76.52250  
4               Nequi                   1   6.25184 -75.56359

In [3]:
tienda.shape

(2359, 12)

# 1. Análisis de facturación

## 1.1 Ingreso total por tienda

In [4]:
ingreso_total_T1 = tienda.Precio.sum()
ingreso_total_T1

1150880400.0

In [5]:
ingreso_total_T2 = tienda2.Precio.sum()
ingreso_total_T2

1116343500.0

In [6]:
ingreso_total_T3 = tienda3.Precio.sum()
ingreso_total_T3

1098019600.0

In [7]:
ingreso_total_T4 = tienda4.Precio.sum()
ingreso_total_T4

1038375700.0

> La tienda 4 es la que menos ingresos ha tenido

# 2. Ventas por categoría

In [8]:
tienda['Categoría del Producto'].unique()

array(['Electrónicos', 'Muebles', 'Juguetes', 'Electrodomésticos',
       'Artículos para el hogar', 'Deportes y diversión', 'Libros',
       'Instrumentos musicales'], dtype=object)

In [ ]:
# total de venta por producto tienda 1
tienda.groupby('Categoría del Producto')['Precio'].sum()

Categoría del Producto
Artículos para el hogar     12698400.0
Deportes y diversión        39290000.0
Electrodomésticos          363685200.0
Electrónicos               429493500.0
Instrumentos musicales      91299000.0
Juguetes                    17995700.0
Libros                       8784900.0
Muebles                    187633700.0
Name: Precio, dtype: float64

In [ ]:
# cantidad de productos vendidos por categoria de la tienda 1
conteo_T1 = tienda['Categoría del Producto'].value_counts().reset_index()
conteo_T1.columns = ['Categoria', 'Cantidad']
conteo_T1

Categoria  Cantidad
0                  Muebles       465
1             Electrónicos       448
2                 Juguetes       324
3        Electrodomésticos       312
4     Deportes y diversión       284
5   Instrumentos musicales       182
6                   Libros       173
7  Artículos para el hogar       171

In [12]:
# cantidad de productos vendidos por categoria de la tienda 2
conteo_T2 = tienda2['Categoría del Producto'].value_counts().reset_index()
conteo_T2.columns = ['Categoria', 'Cantidad']
conteo_T2

Categoria  Cantidad
0                  Muebles       442
1             Electrónicos       422
2                 Juguetes       313
3        Electrodomésticos       305
4     Deportes y diversión       275
5   Instrumentos musicales       224
6                   Libros       197
7  Artículos para el hogar       181

In [13]:
# cantidad de productos vendidos por categoria de la tienda 3
conteo_T3 = tienda3['Categoría del Producto'].value_counts().reset_index()
conteo_T3.columns = ['Categoria', 'Cantidad']
conteo_T3

Categoria  Cantidad
0                  Muebles       499
1             Electrónicos       451
2                 Juguetes       315
3        Electrodomésticos       278
4     Deportes y diversión       277
5                   Libros       185
6   Instrumentos musicales       177
7  Artículos para el hogar       177

In [14]:
# cantidad de productos vendidos por categoria de la tienda 1
conteo_T4 = tienda4['Categoría del Producto'].value_counts().reset_index()
conteo_T4.columns = ['Categoria', 'Cantidad']
conteo_T4

Categoria  Cantidad
0                  Muebles       480
1             Electrónicos       451
2                 Juguetes       338
3     Deportes y diversión       277
4        Electrodomésticos       254
5  Artículos para el hogar       201
6                   Libros       187
7   Instrumentos musicales       170

> Las 3 categorias mas vendidas de todas las tiendas son: Muebles, electronicos y juguetes unidades.

> La tienda 3 posee mayor cantidad de ventas de muebles con un total de 499 unidades.

> Las tiendas 3 y 4 han vendido mayor cantidad de electronico con un total de 451 unidades.

> La tienda 4 ha vendido mayor cantidad de juguetes con un total de 338 unidades

# 3. Calificación promedio de la tienda


# 4. Productos más y menos vendidos

# 5. Envío promedio por tienda